modules/phase_field/src/action/RigidBodyMultiKernelAction.C

```diff
@@ -72,62 +76,70 @@ RigidBodyMultiKernelAction::act()
+      params.set<MaterialPropertyName>("mob_name") = getParam<MaterialPropertyName>("mob_name");
 
       std::string kernel_name = "ACInt_" + var_name;
-      _problem->addKernel("ACInterface", kernel_name, parameters);
+      _problem->addKernel("ACInterface", kernel_name, params);
     }
 
     //
     // Set up the AllenCahn kernels
     //
 
     {
-      InputParameters parameters = _factory.getValidParams("AllenCahn");
-      parameters.set<NonlinearVariableName>("variable") = var_name;
-      parameters.set<std::vector<VariableName> >("args") = v;
-      parameters.set<MaterialPropertyName>("mob_name") = getParam<MaterialPropertyName>("mob_name");
-      parameters.set<MaterialPropertyName>("f_name") = getParam<MaterialPropertyName>("f_name");
-      parameters.set<bool>("implicit") = _implicit;
-      parameters.set<bool>("use_displaced_mesh") = getParam<bool>("use_displaced_mesh");
+      InputParameters params = _factory.getValidParams("AllenCahn");
+      params.set<NonlinearVariableName>("variable") = var_name;
+      params.set<std::vector<VariableName> >("args") = v;
+      params.set<MaterialPropertyName>("mob_name") = getParam<MaterialPropertyName>("mob_name");
+      params.set<MaterialPropertyName>("f_name") = getParam<MaterialPropertyName>("f_name");
+      params.set<bool>("implicit") = _implicit;
+      params.set<bool>("use_displaced_mesh") = getParam<bool>("use_displaced_mesh");
 
       std::string kernel_name = "AC_" + var_name;
-      _problem->addKernel("AllenCahn", kernel_name, parameters);
+      _problem->addKernel("AllenCahn", kernel_name, params);
     }
 
     //
     // Set up SingleGrainRigidBodyMotion kernels
     //
 
     {
-      InputParameters parameters = _factory.getValidParams("SingleGrainRigidBodyMotion");
-      parameters.set<NonlinearVariableName>("variable") = var_name;
-      parameters.set<std::vector<VariableName> >("v") = v;
-      parameters.set<std::vector<VariableName> >("c") = {getParam<VariableName>("c")};
-      parameters.set<bool>("implicit") = _implicit;
-      parameters.set<bool>("use_displaced_mesh") = getParam<bool>("use_displaced_mesh");
+      InputParameters params = _factory.getValidParams("SingleGrainRigidBodyMotion");
+      params.set<NonlinearVariableName>("variable") = var_name;
+      params.set<std::vector<VariableName> >("v") = v;
+      params.set<unsigned int>("op_index") = op;
+      params.set<std::vector<VariableName> >("c") = {getParam<VariableName>("c")};
+      if (isParamValid("base_name"))
+        params.set<std::string>("base_name") = {getParam<std::string>("base_name")};
+      params.set<Real>("translation_constant") = {getParam<Real>("translation_constant")};
+      params.set<Real>("rotation_constant") = {getParam<Real>("rotation_constant")};
+      params.set<UserObjectName>("grain_force") = {getParam<UserObjectName>("grain_force")};
+      params.set<UserObjectName>("grain_tracker_object") = {getParam<UserObjectName>("grain_tracker_object")};
```

Comment:
You should probably drop the curly braces in all of these statements. It looks like the new initializer_list syntax but it's not in this context. You just have an anonymous scope which is unecessary here.